controller/users.go

```diff
@@ -49,6 +50,102 @@ func NewUsersController(service *goa.Service, db application.DB, config UsersCon
 		config:     config,
 	}
 }
+func randString(n int) string {
+	const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letterBytes[rand.Int63()%int64(len(letterBytes))]
+	}
+	return string(b)
+}
+
+// Obfuscate runs the obfuscate action to soft delete sensitive data associated
+// with an user and her associated identity.
+func (c *UsersController) Obfuscate(ctx *app.ObfuscateUsersContext) error {
+	isSvcAccount, err := isServiceAccount(ctx, serviceNameAuth)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to determine if account is a service account")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err))
+
+	}
+	if !isSvcAccount {
+		log.Error(ctx, map[string]interface{}{
+			"identity_id": ctx.ID,
+		}, "account used to call create api is not a service account")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(errs.New("a non-service account tried to create a user.")))
+	}
+	u, err := uuid.FromString(ctx.ID)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to convert user id in valid uuid")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrBadRequest(errs.New("invalid user id")))
+
+	}
+	returnResponse := application.Transactional(c.db, func(appl application.Application) error {
+		obfStr := randString(12)
+
+		// Obfuscate User
+		users, err := appl.Users().Query(account.UserFilterByID(u))
+		if err != nil || len(users) != 1 {
+			log.Error(ctx, map[string]interface{}{
+				"user_id": u,
+				"err":     err,
+			}, "unable to load the user")
+			return errs.WithStack(err)
+		}
+		user := users[0]
+		user.Email = obfStr + "@mail.com"
+		user.FullName = obfStr
+		user.ImageURL = obfStr
+		user.Bio = obfStr
+		user.URL = obfStr
+		user.ContextInformation = nil
+		user.Company = obfStr
+		if err := appl.Users().Save(ctx.Context, &user); err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"user_id": u,
+				"err":     err,
+			}, "unable to obfuscate the user")
+			return errs.WithStack(err)
```

Comment:
We should always wrap errors before returning it. It helps in debugging when things go south.
```suggestion
			return errs.Wrap(err, "failed to obfuscate the user")
```